addons/product/models/product_product.py

```diff
@@ -658,6 +659,16 @@ def price_compute(self, price_type, uom=None, currency=None, company=None, date=
 
         return prices
 
+    def _get_base_or_overridden_price(self, price_type):
+        """See `product.template._get_base_or_overridden_price`"""
+        self.ensure_one()
+        if price_type == 'list_price' and self._context.get('record_being_sold'):
+            model_name = DETAILED_TYPE_MODELS.get(self.detailed_type)
+            if not model_name:
+                raise ValueError(f'No model found in `DETAILED_TYPE_MODELS` for detailed type "{self.detailed_type}"')
+            return self.env[model_name].browse(self._context['record_being_sold']).price, True
```

Comment:
I'm afraid we have to be careful with this pattern due to multi-tenancy.
Basically, as there are several db on the same server, and not all have all modules installed, we must be careful when using global variables...
If we still feel that it is a good way to go, we should also check that `model_name` is in `self.env` first.

Also, maybe the error we return in any case (and that the user end may also see) should be a little less technical. Something like "Unable to gather all the information about the product"  ?

Reply:
Thanks, I have changed the error message as suggested.
And as discussed, added a comment saying that model_name must be in self.env as self.detailed_type is provided by the class that define its correspondence with the model through the map DETAILED_TYPE_MODELS. But of course, if it seems still not safe enough, we call get back to the inheritance implementation.

Reply:
up!